r/R/dataset-write.R

```diff
@@ -41,6 +41,9 @@
 #' will yield `"part-0.feather", ...`.
 #' @param hive_style logical: write partition segments as Hive-style
 #' (`key1=value1/key2=value2/file.ext`) or as just bare values. Default is `TRUE`.
+#' @param hive_null_fallback If writing partition segments as Hive-style
```

Comment:
I would either (1) call this `null_fallback` so it matches the name of the arg elsewhere or (2) omit it from the signature and pull it out of `...` in the function below.

Is there null fallback behavior for non-hive?

Reply:
Right now "non-hive" means directory as those are the only two in common usage.  Directory partitioning handles nulls by omitting directories or raising an error (if an outer directory is null but an inner directory is not).  So there is no concept of a "null_fallback"

Reply:
I think omitting is fine.  The line `partitioning <- HivePartitioning$create(partition_schema, hive_null_fallback)` would become `partitioning <- hive_partition(partition_schema)` as `HivePartitioning$create` cannot have a default value for `hive_null_fallback`.

Reply:
Ok I've done some shuffling to make this a possibility